frontend/src/events/game/handlers/startGame.ts

```diff
@@ -1,10 +1,35 @@
+import { lobby } from '@/connection';
+import type { Game } from '@/interfaces/Lobby';
 
-import { lobby } from '@/connection'
-
+const INITIAL_HEALTH = 3;
+const INITIAL_WAIT_TIME = 5;
+const INITIAL_MATCH_NUMBER = 1;
+const INITIAL_ROUND_NUMBER = 1;
 
 /**
  *  Evento indicando o início de um jogo completo de "Little Fuck"
  */
 export function handleStartGame() {
+  if (lobby.value === null) {
+    throw new Error('Você não está em um lobby !');
+  }
+
+  if (lobby.value.game) {
+    throw new Error('Não foi possível iniciar o jogo: O jogo já começou !');
+  }
+
+  const players = lobby.value.players as Array<{ id: string }>;
+  const playersHealth = players.reduce<Record<string, number>>((playersHealth, player) => {
```

Comment:
Achei legal o uso do reduce dessa forma. Algum motivo em especifico para o uso dele aqui, substituindo um "for" convencional ?

Reply:
Sem nenhum motivo em especial por trás. Apenas preferências pessoais.

São raros os casos que prefiro usar um `for` convencional (`for(let i = 0; i< x; i++)`) ao lugar das funções de iteração em arrays do js 

Em especial, utilizo o `.reduce()` quando vou iterar o array e gerar como saída um único objeto/valor, com nesse caso que resulta em `playersHealth`